service_container/tags.rst

```diff
@@ -564,3 +564,242 @@ application handlers.
+
+            $handler_two = $handlers['handler_two']:
+            $handler_one = $handlers['handler_one'];
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, by implementing a static
+    method named like this : ``getDefaultIndexAttributeNameName``
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            services:
+                App\Handler\One:
+                    tags:
+                        - { name: 'app.handler', priority: 20 }
+
+                App\Handler\Two:
+                    tags:
+                        - { name: 'app.handler', index: 'handler_two' }
+
+                App\HandlerCollection:
+                    # inject all services tagged with app.handler as first argument
+                    arguments: [!tagged { tag: 'app.handler', index_by: 'index' }]
+
+        .. code-block:: xml
+
+            <!-- config/services.xml -->
+            <?xml version="1.0" encoding="UTF-8" ?>
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+                <services>
+                    <service id="App\Handler\One">
+                        <tag name="app.handler" priority="20" />
+                    </service>
+
+                    <service id="App\Handler\Two">
+                        <tag name="app.handler" index="handler_two" />
+                    </service>
+
+                    <service id="App\HandlerCollection">
+                        <!-- inject all services tagged with app.handler as first argument -->
+                        <argument type="tagged" tag="app.handler" index-by="index" />
+                    </service>
+                </services>
+            </container>
+
+        .. code-block:: php
+
+            // config/services.php
+            $container->register(App\Handler\One::class)
+                ->addTag('app.handler', ['priority' => 20]);
+
+            $container->register(App\Handler\One::class, ['index' => 'handler_two']);
+
+            $container->register(App\HandlerCollection::class)
+                // inject all services tagged with app.handler as first argument
+                ->addArgument(new TaggedIteratorArgument('app.handler', 'index'));
+
+    And here is how your ``App\Handler\One`` should now be :
+
+    .. code-block:: php
+
+        // src/Handler/One.php
+        namespace App\Handler;
+
+        class One
+        {
+            public static function getDefaultIndexName() {
+                return 'handler_one';
+            }
+        }
+
+    You also can define the name of the static method to implement on each service with the ``default_index_method`` attribute on the argument.
```

Comment:
Same down here: show the PHP code first, then the changes in YAML (and only show the code that actually needs to be changed.